### ISSUE TITLE
How to construct indices on primitives?

### ISSUE THREAD

**Issue by DagerD** [dpisani/gltf-builder#5]
Hello.

Can you please explain what is indices and how to construct them? For example here [https://github.com/dpisani/gltf-builder/blob/master/guides/materials-and-textures.md](url) we use them for texture applied to primitive. What if we have primitive of other form?

I read gltf spec and sources of this project but still don't understand.

Your project is awesome!

**Comment by dpisani** [comment]
Hey @DagerD , thanks for taking interest in gltf-builder!

Indices are a way to re-use the vertices that you define in your primitive. I found these articles about indexing which might help you gain a better understanding, though they are focused on use for OpenGL which makes for dense reading (sorry!)

http://ogldev.atspace.co.uk/www/tutorial10/tutorial10.html
https://openglbook.com/chapter-3-index-buffer-objects-and-primitive-types.html
https://gamedev.stackexchange.com/questions/68838/what-is-the-purpose-of-indices-in-3d-rendering

I also added some more docs on creating geometry which adds a bit more explanation on using indices with this library. https://github.com/dpisani/gltf-builder/blob/master/guides/basic-geometry.md

I hope this helps!
